src/java/org/apache/cassandra/service/accord/AccordSafeCommandsForRanges.java

```diff
@@ -21,61 +21,25 @@
 import java.util.NavigableMap;
 import java.util.Objects;
 
-import accord.primitives.Range;
 import accord.primitives.Ranges;
 import accord.primitives.TxnId;
 import accord.utils.async.AsyncChains;
 import accord.utils.async.AsyncResult;
 import org.apache.cassandra.utils.Pair;
 
-public class AccordSafeCommandsForRanges implements AccordSafeState<Range, CommandsForRanges>
+public class AccordSafeCommandsForRanges extends ImmutableAccordSafeState<Ranges, CommandsForRanges>
```

Comment:
the reason for this was that `revert` failed for ranges; found by the improved test cases.  I could just impl `revert` properly in this class but figured a `Immutable*` class would make things more clear

Reply:
I think most safe states are immutable, aren't they? Should they all be migrated to this new interface?

Reply:
most allow `set` to update.  The data is immutable but the references may be updated; but CommandsForRanges doesn't allow `set`.